### PR TITLE
Add HMatrix::scale method

### PR DESCRIPTION
**Pull request by dbarbier**

### CODE REVIEW

**Comment by jschueller** [comment]
I think the docstring is missing

**Comment by dbarbier** [comment]
Ah yes, I will add it, thanks

File: python/src/HMatrixImplementation_doc.i.in [hunk 1]
```diff
@@ -227,6 +227,19 @@ OT_HMatrix_fullrkRatio_doc
...
 // ---------------------------------------------------------------------
 
+%define OT_HMatrix_scale_doc
+"Scale matrix in-place A=alpha*A.
+
+Parameters
+----------
+alpha : float
+    Coefficient
+%enddef
```
Comment (jschueller) [inline python/src/HMatrixImplementation_doc.i.in]: the ending quote is missing

**Comment by regislebrun** [comment]
+1

**Pull request merged**